### PR TITLE
First pass at getting the `post_tags` widget filter to work.

### PR DESCRIPTION
**Pull request by Camwyn**
First pass at getting the `post_tags` widget filter to work.

Some changes to go along with https://github.com/moderntribe/events-pro/pull/1639

[ECP-597]

[ECP-597]: https://moderntribe.atlassian.net/browse/ECP-597

### CODE REVIEW

File: src/Tribe/Views/V2/View.php [hunk 1]
```diff
@@ -1138,6 +1138,7 @@ protected function filter_template_vars( array $template_vars ) {
 	 * @return array The arguments, ready to be set on the View repository instance.
 	 */
 	protected function setup_repository_args( Context $context = null ) {
+		bdump('view_setup_repository_args');
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php:1141]: * :no_entry_sign: Expected 1 spaces after opening parenthesis; 0 found
* :no_entry_sign: Expected 1 spaces before closing parenthesis; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis

<sub>hash: b906ba7ed9f89ef08b4d4db0374c381c</sub>

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 2]
```diff
@@ -61,6 +62,37 @@ protected function setup_template_vars() {
...
+	 * Sets up the View repository arguments from the View context or a provided Context object.
+	 *
+	 * @since 4.9.3
+	 *
+	 * @param  Context|null $context A context to use to setup the args, or `null` to use the View Context.
+	 *
+	 * @return array The arguments, ready to be set on the View repository instance.
+	 */
+	protected function setup_repository_args( Context $context = null ) {
+		bdump('widget_setup_repository_args');
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:75]: * :no_entry_sign: Expected 1 spaces after opening parenthesis; 0 found
* :no_entry_sign: Expected 1 spaces before closing parenthesis; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis

<sub>hash: dbad249068566b809743b38392432894</sub>

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 3]
```diff
@@ -61,6 +62,37 @@ protected function setup_template_vars() {
...
+		$context = null !== $context ? $context : $this->context;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:76]: :warning: Equals sign not aligned with surrounding assignments; expected 5 spaces but found 1 space

<sub>hash: 0e36dc6e1ee79763ec86060a8095ede8</sub>

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 4]
```diff
@@ -61,6 +62,37 @@ protected function setup_template_vars() {
...
+
+		/**
+		 * A widget-specific filter for repository args, based on widget slug.
+		 * Allows other plugins to add/remove args for the repository pre-query.
+		 *
+		 * @param $args
+		 * @param $context
+		 */
+		$args = apply_filters( "tribe_events_views_v2_{$this->get_slug()}_repository_args", $args, $context, $context_arr );
+		bdump( [
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:88]: :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: ebaeb7952e6686f114ebd36837ee0b0a</sub>

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 5]
```diff
@@ -61,6 +62,37 @@ protected function setup_template_vars() {
...
+			'END_widget_setup_repository_args',
+			$args
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:90]: * :no_entry_sign: Each array item in a multi-line array declaration must end in a comma
* :no_entry_sign: Comma required after last value in array declaration

<sub>hash: 316f344f5436a75c320b71a009840748</sub>

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 6]
```diff
@@ -61,6 +62,37 @@ protected function setup_template_vars() {
...
+		]
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:91]: :no_entry_sign: Multi-line function call not indented correctly; expected 12 spaces but found 8

<sub>hash: 0c78af2880fddb7dd3bc3ac8befc7c16</sub>

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 7]
```diff
@@ -61,6 +62,37 @@ protected function setup_template_vars() {
...
+	);
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:92]: * :no_entry_sign: Multi-line function call not indented correctly; expected 8 spaces but found 4
* :no_entry_sign: Line indented incorrectly; expected at least 2 tabs, found 1

<sub>hash: da32f78babd7296a14ef2e0cf232462d</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 14 error(s)
:warning: 1 warning(s)

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 8]
```diff
@@ -61,6 +62,32 @@ protected function setup_template_vars() {
...
 	}
 
+	/**
+	 * Sets up the View repository arguments from the View context or a provided Context object.
+	 *
+	 * @since 4.9.3
+	 *
+	 * @param  Context|null $context A context to use to setup the args, or `null` to use the View Context.
+	 *
+	 * @return array The arguments, ready to be set on the View repository instance.
```
Comment (lucatume) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:72]:
```suggestion
	 * @return array<string,mixed> The arguments, ready to be set on the View repository instance.
```

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 9]
```diff
@@ -61,6 +62,32 @@ protected function setup_template_vars() {
...
+	 */
+	protected function setup_repository_args( Context $context = null ) {
+		$context     = null !== $context ? $context : $this->context;
+		$context_arr = $context->to_array();
```
Comment (lucatume) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:76]: Nope, let's avoid this as it's costly and inefficient.
Just pass a reference to the context in the filter.

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 10]
```diff
@@ -61,6 +62,32 @@ protected function setup_template_vars() {
...
+		$args        = parent::setup_repository_args( $context );
+
+		/**
+		 * A widget-specific filter for repository args, based on widget slug.
+		 * Allows other plugins to add/remove args for the repository pre-query.
+		 *
+		 * @param array<string,mixed> $args        The arguments, ready to be set on the View repository instance.
+		 * @param array<string,mixed> $context_arr The context to use to setup the args,in array format.
+		 */
+		$args = apply_filters( "tribe_events_views_v2_{$this->get_slug()}_repository_args", $args, $context_arr );
```
Comment (lucatume) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:86]:
```suggestion
		$args = apply_filters( "tribe_events_views_v2_{$this->get_slug()}_repository_args", $args, $context );
```

Just pass a reference to the context, dumping the context to array is inefficient and an eager resolution of an object optimized for JIT resolution.

**Review by lucatume — CHANGES_REQUESTED** [review]

File: src/Tribe/Views/V2/Views/Widgets/Widget_View.php [hunk 11]
```diff
@@ -7,6 +7,7 @@
  */
 
 namespace Tribe\Events\Views\V2\Views\Widgets;
+use Tribe__Context as Context;
 
 use Tribe\Events\Views\V2\View;
```
Comment (paulmskim) [inline src/Tribe/Views/V2/Views/Widgets/Widget_View.php:12]: can we also fix this spacing?

**Review by paulmskim — APPROVED** [review]

**Review by lucatume — APPROVED** [review]

**Pull request merged**